src/level_four/core/controller/receive_transaction_controller.c

```diff
@@ -75,6 +75,7 @@ void receive_transaction_controller()
     switch (flow_level.level_three) {
 
     case RECV_TXN_FIND_XPUB: {
+        memzero(wallet_credential_data.passphrase, sizeof(wallet_credential_data.passphrase));
```

Comment:
lets reset this in `reset_flow_level` as well.

Reply:
Also, add this just after passphrase is used. i.e., after the call to `mnemonic_to_seed`